lint/src/main/java/permissions/dispatcher/NoMatchingNeedsPermissionDetector.java

```diff
@@ -0,0 +1,113 @@
+package permissions.dispatcher;
+
+import com.android.tools.lint.detector.api.Category;
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Implementation;
+import com.android.tools.lint.detector.api.Issue;
+import com.android.tools.lint.detector.api.JavaContext;
+import com.android.tools.lint.detector.api.Scope;
+import com.android.tools.lint.detector.api.Severity;
+
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import lombok.ast.Annotation;
+import lombok.ast.AstVisitor;
+import lombok.ast.CompilationUnit;
+import lombok.ast.ForwardingAstVisitor;
+import lombok.ast.Node;
+
+public class NoMatchingNeedsPermissionDetector extends Detector implements Detector.JavaScanner {
+
+    public static final Issue ISSUE = Issue.create("NoMatchingNeedsPermission",
+            "No matching @NeedsPermission found",
+            "The processor would ignore @OnShowRationale annotation if there is no matching @NeedsPermissions. For instance, if there is @OnShowRationale(Manifest.permission.CAMERA), then there must be @NeedsPermission(Manifest.permission.CAMERA) in the same class.",
```

Comment:
For this short and long description is not good enough for this lint. I'm not sure how to express this lint...